File: CPE-1star/UVA10101-Bangla Numbers .py
#先輸出第一位(數字)，判斷剩下是否為0，才開始先存文字，最後一個一個輸出
sequence=1
while True:
	try:
		n=int(input())
		temp=[]
		if n==0:
			temp.append(0)
		else:
			while n>0:
				try:
					temp2=n%100
					if temp2!=0:
						temp.append(temp2)
					n=n//100
					if n==0:
						break
						
					temp2=n%10	
					if temp2!=0:
						temp.append("shata")
						temp.append(temp2)
					n=n//10
					if n==0:
						break
						
					temp2=n%100	
					if temp2!=0:
						temp.append("hajar")
						temp.append(temp2)
					n=n//100
					if n==0:
						break
						
					temp2=n%100	
					if temp2!=0:
						temp.append("lakh")
						temp.append(temp2)
					n=n//100
					if n==0:
						break
						
					if n!=0:
						temp.append("kuti")
				except:
					break
		temp.reverse()
		if sequence<10:
			print("   %d. "%(sequence),end="")
		else:
			print("  %d. "%(sequence),end="")
		sequence+=1
		
		for i in range(len(temp)-1):
			print(temp[i],end=" ")
		print(temp[-1])
	except EOFError:
		break

File: CPE-1star/UVA10101-Bangla Numbers.cpp
#include<iostream>
#include <vector>
#include <algorithm>

using namespace std;
int main(){
    long long n;
    int sequence=1;
    while(cin>>n){
        vector<string> temp;
        if(n==0)
            temp.push_back("0");
        else{
            int temp2;
            while(n>0){
                temp2=n%100;
                if(temp2!=0)
                    temp.push_back(to_string(temp2));
                n=n/100;
                if(n==0)
                    break;
                temp2=n%10;
                if(temp2!=0){
                    temp.push_back("shata");
                    temp.push_back(to_string(temp2));
                }
                n=n/10;
                if(n==0)
                    break;
                temp2=n%100;	
				if(temp2!=0){
					temp.push_back("hajar");
					temp.push_back(to_string(temp2));
                }
				n=n/100;
				if(n==0)
					break;	
				temp2=n%100;
				if(temp2!=0){
					temp.push_back("lakh");
					temp.push_back(to_string(temp2));
                }
				n=n/100;
				if(n==0)
					break;
				if(n!=0)
				    temp.push_back("kuti");
            }
        }
        reverse(temp.begin(),temp.end());
        if(sequence<10)
            cout<<"   "<<sequence<<". ";
        else    
            cout<<"  "<<sequence<<". ";
        sequence+=1;
        for(int i=0;i<temp.size()-1;i++)
            cout<<temp[i]<<" ";
        cout<<temp.back()<<"\n";
    }
    return 0;
}